### PR TITLE
use original data point is now based on cycle

### LINKED ISSUE #1692: Descriptions not visible in not edit mode
https://user-images.githubusercontent.com/1138156/201063016-d907c160-5e07-4438-8c18-e0687de4342e.mov

### PR DESCRIPTION
**Pull request by rfontanarosa**
close #1692

### CODE REVIEW

File: src/client/pages/Landing/Landing.tsx [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
-    <KeyFindings />
-    <Partners />
-  </>
-)
+const Landing: React.FC = () => {
+  const dispatch = useAppDispatch()
+
+  useEffect(() => {
+    dispatch(AssessmentActions.getCountries())
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (minotogna) [inline src/client/pages/Landing/Landing.tsx]: you can dispatch to the dependency and remove the eslint disable ?

File: src/client/store/assessment/actions/initApp.ts [hunk 2]
```diff
@@ -11,9 +11,12 @@ export const initApp = createAsyncThunk<
     countries: Array<Country>
     regionGroups: Record<string, RegionGroup>
   },
-  void
->('assessment/get/init', async () => {
-  const { data } = await axios.get(ApiEndPoint.init())
+  {
+    assessmentName: string
+    cycleName: string
+  } | void
```
Comment (minotogna) [inline src/client/store/assessment/actions/initApp.ts]: is this needed? or can it be rolled back to void ?

File: src/server/api/area/index.ts [hunk 3]
```diff
@@ -4,11 +4,13 @@ import { ApiEndPoint } from '@meta/api/endpoint'
...
 import { AuthMiddleware } from '@server/middleware/auth'
 
+import { getCountries } from './getCountries'
 import { postCountry } from './postCountry'
 
 export const AreaApi = {
   init: (express: Express): void => {
     // Country
     express.post(ApiEndPoint.Area.country(), AuthMiddleware.requireEditTableData, postCountry)
+    express.get(ApiEndPoint.Area.countries(), getCountries)
```
Comment (minotogna) [inline src/server/api/area/index.ts]: can u try requireView auth middleware ?

File: src/server/api/init/init.ts [hunk 4]
```diff
@@ -8,21 +8,17 @@ import { SettingsController } from '@server/controller/settings'
...
-    const props = name ? { assessmentName: name } : { id: settings.defaultAssessmentId }
-    const { assessment, cycle } = await AssessmentController.getOneWithCycle(props)
+    const props = assessmentName ? { assessmentName } : { id: settings.defaultAssessmentId }
+    const { assessment, cycle } = await AssessmentController.getOneWithCycle({ ...props, cycleName })
 
-    const [countries, regionGroups] = await Promise.all([
-      AreaController.getCountries({ assessment, cycle }),
-      AreaController.getRegionGroups({ assessment, cycle }),
-    ])
+    const regionGroups = await AreaController.getRegionGroups({ assessment, cycle })
```
Comment (minotogna) [inline src/server/api/init/init.ts]: this should be added to the getCountries endpoint. at this point it can be called getAreas. what do you think?

Comment (rfontanarosa) [reply src/server/api/init/init.ts]: why not

File: src/server/api/area/index.ts (hunk 3 above)
Comment (rfontanarosa) [reply src/server/api/area/index.ts]: it is used by Landing page too, and it is visible to unknown users

File: src/server/api/init/init.ts (hunk 4 above)
Comment (rfontanarosa) [reply src/server/api/init/init.ts]: or maybe initAreas... but getAreas is clear enough

File: src/client/store/assessment/actions/initApp.ts [hunk 5]
```diff
@@ -2,18 +2,15 @@ import { createAsyncThunk } from '@reduxjs/toolkit'
...
 export const initApp = createAsyncThunk<
   {
     assessment: Assessment
-    countries: Array<Country>
-    regionGroups: Record<string, RegionGroup>
   },
   void
->('assessment/get/init', async () => {
-  const { data } = await axios.get(ApiEndPoint.init())
+>('assessment/get/init', async (params) => {
```
Comment (sorja) [inline src/client/store/assessment/actions/initApp.ts:10]: params type

File: src/client/store/assessment/slice.ts [hunk 6]
```diff
@@ -11,13 +9,19 @@ export const assessmentSlice = createSlice<AssessmentState, SliceCaseReducers<As
   reducers: {},
   extraReducers: (builder) => {
     builder.addCase(initApp.fulfilled, (state, { payload }) => {
-      const { assessment, countries, regionGroups } = payload
+      const { assessment } = payload
 
       state.assessment = assessment
```
Comment (sorja) [inline src/client/store/assessment/slice.ts:14]: should we remove countries and regionGroups from initApp if we don't use them here?

File: src/server/api/area/index.ts [hunk 7]
```diff
@@ -4,11 +4,13 @@ import { ApiEndPoint } from '@meta/api/endpoint'
...
 import { AuthMiddleware } from '@server/middleware/auth'
 
+import { getAreas } from './getAreas'
 import { postCountry } from './postCountry'
 
 export const AreaApi = {
   init: (express: Express): void => {
     // Country
     express.post(ApiEndPoint.Area.country(), AuthMiddleware.requireEditTableData, postCountry)
+    express.get(ApiEndPoint.Area.areas(), getAreas)
```
Comment (minotogna) [inline src/server/api/area/index.ts:14]: Auth Middleware

**Review by minotogna — CHANGES_REQUESTED** [review]

File: src/server/api/area/index.ts (hunk 7 above)
Comment (rfontanarosa) [reply src/server/api/area/index.ts:14]: getCountries is used by the Landing page too

this means that anonymous user are able to get the country list

**Review by minotogna — APPROVED** [review]

**Review by sorja — APPROVED** [review]

**Pull request merged**